bin/main.ml

```diff
@@ -61,17 +61,33 @@ module Prompt = struct
     | true -> Lwt_io.printl "I: reading from stdin. Please type the secret and then do Ctrl+d twice to terminate input"
     | false -> Lwt.return_unit
 
-  let read_input_from_stdin () = Lwt_io.(read stdin)
-
-  let rec input_and_validate_loop ?(transform = fun x -> x) get_secret_input =
-    let%lwt input = get_secret_input () in
-    let secret = transform input in
+  let read_input_from_stdin ?initial:_ () = Lwt_io.(read stdin)
+
+  let rec input_and_validate_loop ?(transform = fun x -> x) ?initial get_secret_input =
+    let remove_trailing_newlines s =
+      let rec loop s =
+        match String.ends_with ~suffix:"\n" s with
+        | true -> loop (String.sub s 0 (String.length s - 1))
+        | false -> s
+      in
+      loop s
+    in
+    let%lwt input = get_secret_input ?initial () in
+    let input = transform input in
+    (* Remove bash commented lines from the secret and any trailing newlines *)
+    let secret =
+      String.split_on_char '\n' input
+      |> List.filter (fun line -> not (String.starts_with ~prefix:"#" line))
+      |> String.concat "\n"
+      |> remove_trailing_newlines
+    in
     match Secret.Validation.validate secret with
     | Error (e, _typ) ->
       if is_TTY = false then Shell.die "This secret is in an invalid format: %s" e
       else (
         let%lwt () = Lwt_io.printlf "\nThis secret is in an invalid format: %s" e in
-        if%lwt yesno "Edit again?" then input_and_validate_loop get_secret_input else Lwt.return_error e)
+        if%lwt yesno "Edit again?" then input_and_validate_loop ?initial:(Some input) get_secret_input
+        else Lwt.return_error e)
```

Comment:
Why not `Shell.die` here?

The error is potentially silently ignored later (`create_new_secret` calls `Result.value`).

Reply:
good call, updated the logic. Still returning a `Result` type, but handling it better in the `Edit.edit_secret` function downstream.